File: dofun/safe/deviceinfo/utils/DensityUtils.java
package com.dofun.safe.deviceinfo.utils;

import android.app.Activity;
import android.content.Context;
import android.content.res.Resources;
import android.os.Build;
import android.util.DisplayMetrics;
import android.util.TypedValue;
import android.view.Display;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;

/* loaded from: classes6.dex */
public class DensityUtils {
    public static int getDensityDpi(Context context) {
        DisplayMetrics displayMetrics = context.getResources().getDisplayMetrics();
        return displayMetrics.densityDpi;
    }

    public static float getDensity(Context context) {
        DisplayMetrics displayMetrics = context.getResources().getDisplayMetrics();
        return displayMetrics.density;
    }

    public static int getScreenWidth(Context context) {
        DisplayMetrics display = context.getResources().getDisplayMetrics();
        return display.widthPixels;
    }

    public static int getScreenHeight(Context context) {
        DisplayMetrics display = context.getResources().getDisplayMetrics();
        return display.heightPixels;
    }

    public static int getScreenWidthWithDp(Context context) {
        DisplayMetrics displayMetrics = context.getResources().getDisplayMetrics();
        float density = displayMetrics.density;
        return (int) ((displayMetrics.widthPixels / density) + 0.5f);
    }

    public static int getScreenHeightWithDp(Context context) {
        DisplayMetrics displayMetrics = context.getResources().getDisplayMetrics();
        float density = displayMetrics.density;
        return (int) ((displayMetrics.heightPixels / density) + 0.5f);
    }

    public static int getRefreshRate(Context context) {
        WindowManager windowManager;
        Display display;
        if (context == null || (windowManager = (WindowManager) context.getSystemService("window")) == null) {
            return 0;
        }
        if (Build.VERSION.SDK_INT >= 30) {
            display = context.getDisplay();
        } else {
            display = windowManager.getDefaultDisplay();
        }
        float refreshRate = display.getRefreshRate();
        return (int) refreshRate;
    }

    public static int getStatusBarHeight(Context context) {
        Resources resources = context.getResources();
        int resourceId = resources.getIdentifier("status_bar_height", "dimen", "android");
        return resources.getDimensionPixelSize(resourceId);
    }

    public static int getNavigationBarHeight(Context context) {
        Resources resources = context.getResources();
        int resourceId = resources.getIdentifier("navigation_bar_height", "dimen", "android");
        return resources.getDimensionPixelSize(resourceId);
    }

    public static boolean hideStatusBar(Context context) {
        return checkFullScreenByTheme(context) || checkFullScreenByCode(context) || checkFullScreenByCode2(context);
    }

    private static boolean checkFullScreenByTheme(Context context) {
        Resources.Theme theme = context.getTheme();
        if (theme != null) {
            TypedValue typedValue = new TypedValue();
            boolean result = theme.resolveAttribute(16843277, typedValue, false);
            if (result) {
                typedValue.coerceToString();
                return typedValue.type == 18 && typedValue.data != 0;
            }
        }
        return false;
    }

    private static boolean checkFullScreenByCode(Context context) {
        Window window;
        if (!(context instanceof Activity) || (window = ((Activity) context).getWindow()) == null) {
            return false;
        }
        View decorView = window.getDecorView();
        return (decorView.getSystemUiVisibility() & 4) == 4;
    }

    private static boolean checkFullScreenByCode2(Context context) {
        return (context instanceof Activity) && (((Activity) context).getWindow().getAttributes().flags & 1024) == 1024;
    }

    public static boolean hasNavigationBar(Context context) {
        if (!(context instanceof Activity)) {
            return false;
        }
        WindowManager windowManager = ((Activity) context).getWindowManager();
        Display d = windowManager.getDefaultDisplay();
        DisplayMetrics realDisplayMetrics = new DisplayMetrics();
        d.getRealMetrics(realDisplayMetrics);
        int realHeight = realDisplayMetrics.heightPixels;
        int realWidth = realDisplayMetrics.widthPixels;
        DisplayMetrics displayMetrics = new DisplayMetrics();
        d.getMetrics(displayMetrics);
        int displayHeight = displayMetrics.heightPixels;
        int displayWidth = displayMetrics.widthPixels;
        return realWidth - displayWidth > 0 || realHeight - displayHeight > 0;
    }
}

File: app/src/main/java/com/dofun/safe/deviceinfo/utils/DensityUtils.kt
package com.dofun.safe.deviceinfo.utils

import android.app.Activity
import android.content.Context
import android.content.res.Resources
import android.os.Build
import android.util.DisplayMetrics
import android.util.TypedValue
import android.view.Display
import android.view.View
import android.view.Window
import android.view.WindowManager

object DensityUtils {
    fun getDensityDpi(context: Context): Int {
        val displayMetrics = context.resources.displayMetrics
        return displayMetrics.densityDpi
    }

    fun getDensity(context: Context): Float {
        val displayMetrics = context.resources.displayMetrics
        return displayMetrics.density
    }

    fun getScreenWidth(context: Context): Int {
        val display = context.resources.displayMetrics
        return display.widthPixels
    }

    fun getScreenHeight(context: Context): Int {
        val display = context.resources.displayMetrics
        return display.heightPixels
    }

    fun getScreenWidthWithDp(context: Context): Int {
        val displayMetrics = context.resources.displayMetrics
        val density = displayMetrics.density
        return ((displayMetrics.widthPixels / density) + 0.5f).toInt()
    }

    fun getScreenHeightWithDp(context: Context): Int {
        val displayMetrics = context.resources.displayMetrics
        val density = displayMetrics.density
        return ((displayMetrics.heightPixels / density) + 0.5f).toInt()
    }

    fun getRefreshRate(context: Context?): Int {
        val windowManager = context?.getSystemService("window") as? WindowManager ?: return 0
        val display: Display = if (Build.VERSION.SDK_INT >= 30) {
            context.display
        } else {
            windowManager.defaultDisplay
        }
        val refreshRate = display.refreshRate
        return refreshRate.toInt()
    }

    fun getStatusBarHeight(context: Context): Int {
        val resources: Resources = context.resources
        val resourceId = resources.getIdentifier("status_bar_height", "dimen", "android")
        return resources.getDimensionPixelSize(resourceId)
    }

    fun getNavigationBarHeight(context: Context): Int {
        val resources: Resources = context.resources
        val resourceId = resources.getIdentifier("navigation_bar_height", "dimen", "android")
        return resources.getDimensionPixelSize(resourceId)
    }

    fun hideStatusBar(context: Context): Boolean {
        return checkFullScreenByTheme(context) || checkFullScreenByCode(context) || checkFullScreenByCode2(context)
    }

    private fun checkFullScreenByTheme(context: Context): Boolean {
        val theme = context.theme
        if (theme != null) {
            val typedValue = TypedValue()
            val result = theme.resolveAttribute(16843277, typedValue, false)
            if (result) {
                typedValue.coerceToString()
                return typedValue.type == 18 && typedValue.data != 0
            }
        }
        return false
    }

    private fun checkFullScreenByCode(context: Context): Boolean {
        val window: Window = (context as? Activity)?.window ?: return false
        val decorView: View = window.decorView
        return decorView.systemUiVisibility and 4 == 4
    }

    private fun checkFullScreenByCode2(context: Context): Boolean {
        val activity = context as? Activity ?: return false
        return activity.window.attributes.flags and 1024 == 1024
    }

    fun hasNavigationBar(context: Context): Boolean {
        val activity = context as? Activity ?: return false
        val windowManager = activity.windowManager
        val display = windowManager.defaultDisplay
        val realDisplayMetrics = DisplayMetrics()
        display.getRealMetrics(realDisplayMetrics)
        val realHeight = realDisplayMetrics.heightPixels
        val realWidth = realDisplayMetrics.widthPixels
        val displayMetrics = DisplayMetrics()
        display.getMetrics(displayMetrics)
        val displayHeight = displayMetrics.heightPixels
        val displayWidth = displayMetrics.widthPixels
        return realWidth - displayWidth > 0 || realHeight - displayHeight > 0
    }
}
